core/system.h

```diff
@@ -19,7 +20,21 @@ namespace dronecore {
 class DroneCoreImpl;
 class PluginImplBase;
 
-class Device
+
+/**
+ * @brief This represents GCS DroneCore client.
+ */
+struct Self {
+    static constexpr uint8_t system_id = 0;
+    static constexpr uint8_t component_id = MAV_COMP_ID_SYSTEM_CONTROL;
+    static constexpr MAV_TYPE type = MAV_TYPE_GCS;
+};
+
+/**
```

Comment:
This needs work. How about:

> This class represents a MAVLink system, made up of one or more components (e.g. autopilot, cameras, servos, gimbals, etc). Commonly `System` objects are used to interact with UAVs (including their components) and standalone cameras.

Anything else we need to say?

Reply:
Yeah, I just added whatever came to my mind. I think yours is better.